snmp/datadog_checks/snmp/snmp.py

```diff
@@ -71,7 +71,10 @@ def __init__(self, name, init_config, agentConfig, instances):
     def _load_conf(self, instance):
         tags = instance.get("tags", [])
         ip_address = instance["ip_address"]
-        metrics = instance.get('metrics', [])
+        metrics = []
+        if self.init_config is not None:
+            metrics = self.init_config.get('metrics', [])
+        metrics.extend(instance.get('metrics', []))
```

Comment:
Can we have some de-duplication, or at least make the instance queries override those in init_config?

Reply:
I vote for overriding. I think it's the behavior for other checks where it's possible to define instance configs in the init_config.

Reply:
So we can now do both. If `use_global_metrics` is set to `true` then it would extend. However if it is set to `false` it would override. Regarding duplication. I am not going to add this feature. It would be useful but outside the feature request for the PR and i don't want to over engineer it.